pycbc/filter/matchedfilter.py

```diff
@@ -1831,16 +1831,42 @@ def followup_event_significance(ifo, data_reader, bank,
     onsource_end += coinc_threshold
 
     # Calculate how much time needed to calculate significance
-    trim_pad = (data_reader.trim_padding * data_reader.strain.delta_t)
-    bdur = int(lookback + 2.0 * trim_pad + length_in_time)
-    if bdur > data_reader.strain.duration * .75:
-        bdur = data_reader.strain.duration * .75
+    trim_pad = data_reader.trim_padding * data_reader.strain.delta_t
+    bdur = int(lookback + 2 * trim_pad + length_in_time)
+    bdur_samples = bank.round_up(bdur * bank.sample_rate)
+    max_safe_bdur_samples = int(
+        data_reader.strain.duration * .9 * bank.sample_rate
+    )
+    if bdur_samples > max_safe_bdur_samples:
+        bdur_samples = max_safe_bdur_samples
+        new_lookback = bdur_samples / bank.sample_rate - (2 * trim_pad + length_in_time)
+        if new_lookback > 0:
```

Comment:
Maybe zero is too permissive here, maybe I should set it to the duration of the onsource window?

Reply:
I'd probably do onsource * some factor.